### DISCUSSION TITLE
Understanding Livewire.hook('morph')

### DISCUSSION THREAD

**Discussion by pockie** [livewire/livewire#7596] [1. Help] [answered]
### Livewire version
v3.3.0

### Laravel version
v.10.35.0

### Which PHP version are you using?
PHP 8.2

### Problem
I want to achieve the follwoing. After updating a property and rendering the changes in the browser I want that an external Javascript file is triggered. This Javascript uses Konva.js to draw something.

First I tried sending an event and recieving it via `
$wire.on('post-created', () => {});` from my component when updated() is triggered. But I found out that the event is triggered before the HTML is rendered in browser.

Now I'm trying to do it with `Livewire.hook('morph.updated, () => {})`. This works and after the HTML is rerendered in the browser my drawing with konva.js is updated. The problem is that morph.updated is triggered several times. 

Here is a simple example:
My component:
```php
<?php

namespace App\Livewire;

use Livewire\Component;

class DeckConstruction extends Component
{
    public float $test = 5000;

    public function render()
    {
        return view('livewire.deck-construction');
    }
}
```
My view:
```javascript
<div>
    <div>
        <label>
            <input value="{{ $test }}" type="text" wire:model.live.blur="test"> Test
        </label>
    </div>
    {{ $test }}
    @script
    <script>
        Livewire.hook('morph.updating', () => {
            console.log('morph.updating')
        })
        Livewire.hook('morph.updated', () => {
            console.log('morph.updated');
        })
    </script>
    @endscript
</div>
```
The result in chrome:
![grafik](https://github.com/livewire/livewire/assets/7887953/b87641a4-6d1a-4b25-94e7-e003e6c87c55)

My questions:
* If I don't add `Livewire.hook('morph.updating')};`  `Livewire.hook('morph.updated')};` is not triggered. Why do I need updating to getting updated triggered?
* In this example `Livewire.hook('morph.updated')};` is triggered four times. Why is it triggered four times and not oly one time?
* Is my approach correct or is there a better way to trigger an external script after the HTML is rerendered in the browser?

**Comment by pockie** [comment, marked as answer]
I solved the problem by using the javascript MutationObserver.

**Reply by hetparekh21** [reply]
can you please explain how you solved it , a code example would be great
